vars/buildArchive.groovy

```diff
@@ -2,13 +2,9 @@ void call(Map args = [:]) {
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
     buildManifest(args)
 
-    echo "Archiving into zip: ${args.distribution}/builds/**, ${args.inputManifest}"
-
-    zip(
-        zipFile: "${args.distribution}/archived-builds.zip",
-        archive: true,
-        glob: "${args.distribution}/builds/**"
-    )
+    String stashName = "build-archive-${args.platform}-${args.architecture}-${args.distribution}-${JOB_NAME}-${BUILD_NUMBER}"
```

Comment:
It is error prone to construct the name in two files. Better to have this name construction logic in a dedicated Groovy and both places can call it.

Reply:
I change it to jenkinsfile args pushing the stashName, not bound to the lib file anymore.
Thanks.